satcom/openlst/client_packet_lib.py

```diff
@@ -56,35 +56,42 @@ def from_bytes(cls, bs: bytearray):
 class ClientPacket():
     def __init__(self, data: bytearray, header=None):
         self.header = header or ClientPacketHeader()
-        self.header.length = CLIENT_PACKET_HEADER_LENGTH + len(data)
+        self.header.length = CLIENT_PACKET_HEADER_LENGTH + len(data) - 1
         self._data = data
 
     @property
     def data(self):
-        """Protects data from being modified after instance call"""
+        """Protects data from being modified after instantiation"""
         return self._data
 
     def err(self):
         """Throws an error if any params are out of bounds"""
         if self.header.err() is not None:
             return self.header.err()
-        if self.header.length != CLIENT_PACKET_HEADER_LENGTH + len(self.data):
+        if self.header.length != CLIENT_PACKET_HEADER_LENGTH + len(self.data) - 1:
             return ValueError('packet length unequal to header length')
         return None
 
     def to_bytes(self):
-        """Encodes client packet header and data into bytes"""
+        """Encodes client packet and data into bytes"""
         buf = bytearray(CLIENT_PACKET_HEADER_LENGTH)
-        buf = self.header.to_bytes()
+        buf[:CLIENT_PACKET_HEADER_LENGTH] = self.header.to_bytes()
+        buf[0] += 1
```

Comment:
This looks suspicious. This would be increasing the length field by 1. You should not have to do this, and instead you need to update the tests to expect a lesser value.